Data/Versions.hs

```diff
@@ -850,7 +850,20 @@ semver'' = SemVer <$> majorP <*> minorP <*> patchP <*> optional releaseP <*> opt
 
 -- | Parse a group of digits, which can't be lead by a 0, unless it is 0.
 unsignedP :: Parsec Void Text Word
-unsignedP = (0 <$ char '0') <|> decimal
+unsignedP = asWord64 >>= convertOrFail
+  where
+    asWord64 :: Parsec Void Text Word64
+    asWord64 = (0 <$ char '0') <|> L.decimal
+
+    convertOrFail :: Word64 -> Parsec Void Text Word
+    convertOrFail w | w'' /= w = fail "You are not on a 64-bit system and have encountered a number too large to parse."
+                    | otherwise = pure w'
+      where
+        w' :: Word
+        w' = fromIntegral w
+
+        w'' :: Word64
+        w'' = fromIntegral w'
```

Comment:
When you call `fromIntegral` on a value too large for its target type, it gets truncated, not rolled over. So, it would become equal to the `maxBound` of whatever the target type is.